client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -27,17 +27,22 @@ def unwrap(func: Callable) -> Callable:
 def modules_in_function_body(func: Callable) -> Set[str]:
     """Return a set of all base modules called within the body of the provided function."""
     _func = unwrap(func)
-    _globals = [
+    _global_modules = {
         value.__name__.split('.')[0]  # strip off submodules and classes
         for key, value in inspect.getclosurevars(_func).globals.items()
         if isinstance(value, ModuleType)
-    ]
-    _non_locals = [
-        value.__name__.split('.')[0]  # strip off submodules and classes
-        for key, value in inspect.getclosurevars(_func).nonlocals.items()
-        if isinstance(value, ModuleType)
-    ]
-    return set(_globals + _non_locals)
+    }
+    _global_classes = {
+        inspect.getmodule(value).__name__.split('.')[0]  # strip off submodules
+        for key, value in inspect.getclosurevars(_func).globals.items()
+        if inspect.isclass(value)
```

Comment:
How about functions?

```python
from verta._internal_utils import model_dependencies as md

from numpy import array
from pandas import DataFrame

class Model():
    def predict(self):
        arr = array([1, 2, 3])
        df = DataFrame(arr)
        return df

assert md.class_module_names(Model) == {"pandas"}  # no numpy
```

Could we try doing something more general, if possible? Data scientists could import anything from libraries.

Reply:
<img width="874" alt="image" src="https://user-images.githubusercontent.com/114943931/224770693-98de39b1-8912-43b0-bd7c-b6e411b32369.png">
 Apparently, when a function is used from a 3rd party module like this, the value of `type()` is "class 'builtin_function_or_method'".  However,  the inspect functions `isclass()`, `isfunction()` and `is method()`. all return `False`.  Only `inspect.isbuiltin()` returns `True`.

Reply:
Ok, I think I've found a way to make this simpler and more generic by excluding only `unbound` findings from [getclosurevars](https://docs.python.org/3/library/inspect.html#inspect.getclosurevars) which allows everything else to be caught.  That function sorts out things that could not be resolved (into the 'unbound` category), then [getmodule](https://docs.python.org/3/library/inspect.html#inspect.getmodule) is already set up to handle the various types.